packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,339 @@
+    const topLeft = <Types.Point3>[...imagePositionPatient];
+    const topRight = <Types.Point3>[...imagePositionPatient];
+    const bottomLeft = <Types.Point3>[...imagePositionPatient];
+    const bottomRight = <Types.Point3>[...imagePositionPatient];
+
+    vec3.scaleAndAdd(
+      topRight,
+      imagePositionPatient,
+      columnCosines,
+      columns * columnPixelSpacing
+    );
+    vec3.scaleAndAdd(
+      bottomLeft,
+      imagePositionPatient,
+      rowCosines,
+      rows * rowPixelSpacing
+    );
+
+    vec3.scaleAndAdd(
+      bottomRight,
+      bottomLeft,
+      columnCosines,
+      columns * columnPixelSpacing
+    );
+
+    // check if the topLeft and bottomLeft line is parallel to the viewUp
+    const pointSet1 = [topLeft, bottomLeft, topRight, bottomRight];
+    const pointSet2 = [topLeft, topRight, bottomLeft, bottomRight];
+
+    return { pointSet1, pointSet2 };
+  };
+
+  /**
+   * it is used to draw the length annotation in each
+   * request animation frame. It calculates the updated cached statistics if
+   * data is invalidated and cache it.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   */
+  renderAnnotation = (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ): boolean => {
+    let renderStatus = false;
+    if (!this.configuration.sourceImageIds?.length) {
+      return renderStatus;
+    }
+
+    const { viewport: targetViewport, FrameOfReferenceUID } = enabledElement;
+    const targetImageIds = targetViewport.getImageIds();
+    if (targetImageIds.length < 2) {
+      return renderStatus;
+    }
+
+    const annotations = getAnnotations(this.getToolName(), FrameOfReferenceUID);
+    if (!annotations) {
+      return renderStatus;
+    }
+    const annotation = annotations[0];
+    const { annotationUID } = annotation;
+
+    const sourceImageIds = this.configuration.sourceImageIds;
+    const { focalPoint, viewPlaneNormal } = targetViewport.getCamera();
+
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+    const imageIdNormal = <Types.Point3>(
+      this.getImageIdNormal(sourceImageIds[0])
+    );
+
+    if (this.isParallel(viewPlaneNormal, imageIdNormal)) {
+      // If the source and target viewports are parallel, we don't need to render
+      return renderStatus;
+    }
+
+    const targetViewportPlane = csUtils.planar.planeEquation(
+      viewPlaneNormal,
+      focalPoint
+    );
+
+    for (let i = 0; i < sourceImageIds.length; i++) {
+      // check if pointSets for the imageId was calculated. If not calculate and store
+      if (!annotation.data.pointSets[i]) {
+        annotation.data.pointSets[i] = this.calculateImageIdPointSets(
+          sourceImageIds[i]
+        );
+      }
```

Comment:
This block does not need to be here, as it only relys on the imageIds, and not camera. We can perform this on the setToolActive/Enabled. Note that `renderAnnotaiton` runs 60 times per second